cognite/neat/workflows/steps/lib/io_steps.py

```diff
@@ -241,3 +247,124 @@ def run(self, cdf_client: CogniteClient) -> FlowMessage:  # type: ignore[overrid
             data_set_id=dataset_id,
         )
         return FlowMessage(output_text=f"File {self.configs['local.file_name']} uploaded to CDF successfully")
+
+
+class DownloadDataFromRestApiToFile(Step):
+    """
+    This step downloads the response from a REST API and saves it to a file.
+    """
+
+    description = "This step downloads the response from a REST API and saves it to a file."
+    category = "Input/Output"
+    configurables: ClassVar[list[Configurable]] = [
+        Configurable(
+            name="api_url",
+            value="",
+            label="API URL",
+        ),
+        Configurable(
+            name="output_file_path",
+            value="workflows/workflow_name/output.json",
+            label="Output File Path. The path must be relative to the data store path.",
+        ),
+        Configurable(
+            name="http_method",
+            value="GET",
+            label="HTTP Method (GET/POST/PUT)",
+            options=["GET", "POST", "PUT"],
+        ),
+        Configurable(
+            name="auth_mode",
+            value="none",
+            label="Authentication Mode (basic/token/none)",
+            options=["basic", "token", "none"],
+        ),
+        Configurable(
+            name="username",
+            value="",
+            label="Username (for basic auth)",
+        ),
+        Configurable(
+            name="password",
+            value="",
+            label="Password (for basic auth)",
+            type="password",
+        ),
+        Configurable(
+            name="token",
+            value="",
+            label="Token (for token auth)",
+            type="password",
+        ),
+        Configurable(
+            name="response_destination",
+            value="file",
+            label="Destination for the response (file/flow_message/both)",
+            options=["file", "flow_message", "both"],
+        ),
+    ]
+
+    def run(self) -> FlowMessage:
+        api_url = self.configs["api_url"]
```

Comment:
MyPy will complain here as `self.config` is allowed to be None:
```suggestion
        if self.configs is None:
            raise StepNotInitialized(type(self).__name__)
        api_url = self.configs["api_url"]
```